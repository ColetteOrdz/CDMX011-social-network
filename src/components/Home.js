import { onNavigate } from '../routes.js';
//import { toViewSingUp } from './singUp.js';

// export const toViewHome = () => {
//   const mainDiv = document.createElement('div');
//   mainDiv.textContent = 'Regístrate';
//   const buttonRegister = document.createElement('button');
//   buttonRegister.textContent = 'Regístrate';

//   mainDiv.appendChild(buttonRegister);

//   return mainDiv;
// };

// toViewHome();

export const toViewHome = (container) => {
  console.log("estoy en home");
  const html = `
    <section class="container">
        <img class="logo" src="img/PIC&ART.png" alt="logo">
        <p>Share your ideas in the largest art community. Get inspired with different styles and genres around your country.
        </p>
        <input type="button" class="btn_log login" value="LOG IN" />
        <input type="button" class="btn_log signup" value="SIGN UP" id="toCreate" />
        <p2>or</p2>
        <input type="button" class="btn_log google" value="Continue with Google" />
    </section>`;

  container.innerHTML = html
  const toCreate = document.getElementById("toCreate") 
  toCreate.addEventListener('click', (e) => {
    
   e.preventDefault();
   onNavigate('/signUp');
  });
};
