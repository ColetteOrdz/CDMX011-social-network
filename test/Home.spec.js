/**
 * @jest-environment jsdom
 */
/* eslint-disable */

import './globals/firebase.js';
import { toViewHome } from '../src/components/Home.js';

describe("ColecciÃ³n de test en Home", () => {
    document.body.innerHTML=`<div id="root"></div>`;
    
    it("toViewHome should be a function", () => {
        const rootDiv=document.getElementById('root');
        console.log(rootDiv);
        expect(typeof toViewHome).toBe('function');
    });

    it("Should render in Home", () => {
        const rootDiv=document.getElementById('root');
        console.log(rootDiv);
        expect(rootDiv.innerHTML).toMatchSnapshot();
    });

    // it("Should log when Google button is clicked", () => {
    //     const mockLogInGoogle= jest.fn();
    //         mockLogInGoogle.mockImplementation(()=>Promise.resolve()); 

    //         firebase.auth= jest.fn().mockImplementation(()=>({
    //             signInWithGoogle: mockLogInGoogle,
    //         }));
    //         const rootDiv=document.getElementById('root');
    //         toViewHome(rootDiv);

    //         const email='testing@gmail.com';
    //         const password='12345.';

    //         document.getElementById('toGoogle').click();
    //         expect(mockLogInGoogle).toHaveBeenCalledWith(email, password);
    // });
})