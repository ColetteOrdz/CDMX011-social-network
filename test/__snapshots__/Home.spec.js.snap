// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colecci√≥n de test en Home Should render in Home 1`] = `""`;
